tests/unit/mixins/adapter-fetch-test.js

```diff
@@ -15,512 +12,615 @@ import AdapterFetchMixin, {
+  test('parseFetchResponseForError is able to be overwritten to mutate the error payload that gets passed along', function(assert) {
+    assert.expect(1);
+
+    this.errorAdapter = JSONAPIAdapter.extend(AdapterFetchMixin, {
+      _fetchRequest() {
+        const response = new Response(`{ "errors": ["myoneerror"] }`, {
+          status: 422
+        });
+        return Promise.resolve(response);
+      },
+      parseFetchResponseForError() {
+        return {
+          errors: ['myOverWrittenError']
+        };
       }
-    }
-  }).create();
+    }).create();
 
-  const fetchReturn = this.errorAdapter.ajax('/trigger-a-server-error-with-content');
+    const fetchReturn = this.errorAdapter.ajax(
+      '/trigger-a-server-error-with-content'
+    );
 
-  return fetchReturn.catch((body) => {
-    assert.deepEqual(body.errors, ["myOverWrittenError"]);
+    return fetchReturn.catch(body => {
+      assert.deepEqual(body.errors, ['myOverWrittenError']);
+    });
   });
-});
 
-test('able to handle empty error payloads', function(assert) {
-  assert.expect(2);
+  test('able to handle empty error payloads', function(assert) {
+    assert.expect(2);
 
-  this.errorAdapter = JSONAPIAdapter.extend(AdapterFetchMixin, {
-    _fetchRequest() {
-      const response = new Response(``, {status: 500});
-      return Promise.resolve(response);
-    }
-  }).create();
+    this.errorAdapter = JSONAPIAdapter.extend(AdapterFetchMixin, {
+      _fetchRequest() {
+        const response = new Response(``, { status: 500 });
+        return Promise.resolve(response);
+      }
+    }).create();
 
-  const fetchReturn = this.errorAdapter.ajax('/trigger-an-empty-server-error');
+    const fetchReturn = this.errorAdapter.ajax(
+      '/trigger-an-empty-server-error'
+    );
 
-  return fetchReturn.catch((body) => {
-    const error = body.errors[0]
-    assert.equal(error.detail, '');
-    assert.equal(error.status, '500');
+    return fetchReturn.catch(body => {
+      const error = body.errors[0];
+      assert.equal(error.detail, '');
+      assert.equal(error.status, '500');
+    });
   });
-});
-
-test('default fetch hook is correctly called if not overriden', function(assert) {
-  assert.expect(1);
-
-  const fetchReturn = this.JSONAPIAdapter._ajaxRequest({});
-  assert.ok(fetchReturn instanceof Promise);
-});
 
-test('overridden fetch hook is called when supplied', function(assert) {
-  assert.expect(1);
+  test('default fetch hook is correctly called if not overriden', function(assert) {
+    assert.expect(1);
 
-  const newText = 'intercepted';
-  this.JSONAPIAdapter._fetchRequest = () => newText;
-  const fetchReturn = this.JSONAPIAdapter._ajaxRequest({});
-  assert.equal(fetchReturn, newText);
-});
+    const fetchReturn = this.JSONAPIAdapter._ajaxRequest({});
+    // if in pretender with native fetch testing, pretender will polyfill native fetch
+    assert.ok(fetchReturn instanceof Promise || fetchReturn instanceof window.Promise);
```

Comment:
Highlight
This is the only change other than formatting. We have RSVP.Promise but pretender will have the Promise provided specified in `ember-cli-pretender`